### DISCUSSION TITLE
Better GUI for omitting fields

### DISCUSSION THREAD

**Discussion by FelixHerauten** [retorquere/zotero-better-bibtex#2255] [Ideas]
Hi,

Better Bibtex has in Export > Fields an option to omit fields from exporting them to the bibtex-file. But you have to type them as comma-separated list. Therefore you need to know the correct titles of the fields (e.g. I use the German version and don't know them).

Wouldn't it be a great idea, to have a GUI that lets you choose them in a list?

Best regards,
Felix

**Comment by retorquere** [comment]
They're the names of the bibtex fields, which are always in English.

**Comment by FelixHerauten** [comment]
No, in Better Bibtex 6.7.22 for Zotero on Windows, there are no names of the bibtex-fields in the described dialog. You have to know the english names of the fields, but if you use Zotero in German, why should you know that for every field without looking it up. Giving the opportunity to choose them from a dialog would be a usability-improvement.

**Reply by retorquere** [reply]
We're talking about different fields - if you post a debug log, I can give you a more precise description.

**Reply by FelixHerauten** [reply]
How do I make a debug log?

**Reply by retorquere** [reply]
Right-click an item, Better BibTeX - Support Log

**Reply by retorquere** [reply]
Can you also post a screenshot of the item in the support log as it is shown in Zotero

**Reply by retorquere** [reply]
I think this is where the miscommunication comes from

![image](https://user-images.githubusercontent.com/132108/188303085-200cb951-0d4a-48d6-aa77-de04270d9950.png)

exports to 

```
@article{hemida2022,
  title = {Molar Pregnancy with a Coexisting Living Fetus: A Case Series},
  shorttitle = {Molar Pregnancy with a Coexisting Living Fetus},
  author = {Hemida, Reda and Khashaba, Eman and Zalata, Khaled},
  date = {2022-09-03},
  journaltitle = {BMC Pregnancy and Childbirth},
  shortjournal = {BMC Pregnancy Childbirth},
  volume = {22},
  number = {1},
  eprint = {36057566},
  eprinttype = {pmid},
  pages = {681},
  issn = {1471-2393},
  doi = {10.1186/s12884-022-05004-3},
  url = {https://bmcpregnancychildbirth.biomedcentral.com/articles/10.1186/s12884-022-05004-3},
  urldate = {2022-09-04},
  langid = {english}                                                                                                                                                                         }
```

how would you propose omitting `eprint`, `eprinttype`, should that be what you want? There isn't a strict one-to-one mapping between fields visible in Zotero (which will be language dependent) and bibtex fields (which are not), so it's not that these fields are "translated" during export; the omit list names *output* fields, not input fields. If you want to omit certain output, export it, see what field you want to have removed *in the output*, and add that to the omit list.

**Reply by retorquere** [reply]
Oh and also, create an "Artwork" type item, enter something to its "Medium" field , then switch the type to "Audio recording". The Medium field changes its name to "Format". Even in Zotero, these input field names are type-dependent, when they're not in the data.

**Comment by FelixHerauten** [comment]
My suggestion is about the Better BibTex-options. In the options there is no right-click.

**Reply by retorquere** [reply]
I don't understand what you are trying to say here, sorry.

**Reply by FelixHerauten** [reply]
Like I've said in the first post, it is only a suggestion for improving the GUI. When you go to the options of BetterBibtex, then the tab Export and then the tab Fields, there is an option for omitting fields in the export. This option has an empty line which expects comma-separated values. But therefore you have to know the names, but if you use Zotero in German you don't know these names. E.g. Citavi has a dialog that lets you choose them. Wouldn't it be a nice idea to have something like that in the BetterBibtex-options.

**Reply by retorquere** [reply]
But what you are suggesting is a UI to select/deselect *source*-fields, and functionally, the omit option lists *target* fields, which do not always have a one-to-one relationship to source fields.

I don't know how else to explain this. Does Citavi allow you to select *bibtex* (target) fields, **not** Citavi (source) fields, in German? How? `eprint` does not have a Zotero-GUI field, it is a derived field from other fields -- what would you want to select (in German) to exclude this one field?

You know these names by exporting the bibtex and seeing which *bibtex* field you don't want, not by translating a *source* field. We could talk about a "ignore these source fields" but that sounds downright confusing to me as it is hard to predict what output field that affects.
